test/controllers/update/check.your.answers.controller.spec.ts

```diff
@@ -466,7 +489,490 @@ describe("CHECK YOUR ANSWERS controller", () => {
       expect(resp.text).toContain(OVERSEAS_NAME_MOCK);
       expect(resp.text).toContain(COMPANY_NUMBER);
       expect(resp.text).toContain(REMOVE_CHECK_YOUR_ANSWERS_BACK_LINK);
+    });
+
+    test(`renders the ${UPDATE_CHECK_YOUR_ANSWERS_PAGE} page when an added trust is still involved in the overseas entity`, async () => {
+      const trust = {
+        trust_id: "123",
+        trust_name: "wizzz trust",
+        creation_date_day: "31",
+        creation_date_month: "12",
+        creation_date_year: "1999",
+        trust_still_involved_in_overseas_entity: "Yes",
+        unable_to_obtain_all_trust_info: "No"
+      };
+      const appData = {
+        ...APPLICATION_DATA_UPDATE_BO_MOCK,
+        ["trusts"]: [trust]
+      };
+
+      mockGetApplicationData.mockReturnValue(appData);
+      mockIsActiveFeature.mockReturnValueOnce(true).mockReturnValueOnce(true).mockReturnValueOnce(true).mockReturnValueOnce(true);
+      const resp = await request(app).get(UPDATE_CHECK_YOUR_ANSWERS_URL);
+      expect(resp.status).toEqual(200);
+      expect(resp.text).toContain(TRUST_NAME);
+      expect(resp.text).toContain("wizzz trust");
+      expect(resp.text).toContain(TRUST_CREATION_DATE);
+      expect(resp.text).toContain("31");
+      expect(resp.text).toContain("December");
+      expect(resp.text).toContain("1999");
+      expect(resp.text).toContain(TRUSTS_ADDED);
+      expect(resp.text).not.toContain(TRUSTS_REVIEWED);
+      expect(resp.text).toContain(TRUST_INVOLVED);
+      expect(resp.text).not.toContain(TRUST_CEASED_DATE);
+    });
+
+    test(`renders the ${UPDATE_CHECK_YOUR_ANSWERS_PAGE} page when an added trust is not still involved in the overseas entity`, async () => {
+      const trust = {
+        trust_id: "123",
+        trust_name: "wizzz trust",
+        creation_date_day: "31",
+        creation_date_month: "12",
+        creation_date_year: "1999",
+        trust_still_involved_in_overseas_entity: "No",
+        ceased_date_day: "01",
+        ceased_date_month: "02",
+        ceased_date_year: "2004",
+        unable_to_obtain_all_trust_info: "No"
+      };
+      const appData = {
+        ...APPLICATION_DATA_UPDATE_BO_MOCK,
+        ["trusts"]: [trust]
+      };
+
+      mockGetApplicationData.mockReturnValue(appData);
+      mockIsActiveFeature.mockReturnValueOnce(true).mockReturnValueOnce(true).mockReturnValueOnce(true).mockReturnValueOnce(true);
+      const resp = await request(app).get(UPDATE_CHECK_YOUR_ANSWERS_URL);
+      expect(resp.status).toEqual(200);
+      expect(resp.text).toContain(TRUST_NAME);
+      expect(resp.text).toContain("wizzz trust");
+      expect(resp.text).toContain(TRUST_CREATION_DATE);
+      expect(resp.text).toContain("31");
+      expect(resp.text).toContain("December");
+      expect(resp.text).toContain("1999");
+      expect(resp.text).toContain(TRUSTS_ADDED);
+      expect(resp.text).not.toContain(TRUSTS_REVIEWED);
+      expect(resp.text).toContain(TRUST_INVOLVED);
```

Comment:
would it be possible to check the value of TRUST_INVOLVED , not that constant value but it should show "No" on the page. There are probably lots of No's in the text but there might be a string in the resp.text containing the No and something to identify it as the trust involved value.

Reply:
similarly for when that value is Yes, could we check that the Yes appears as the answer for that item.

Reply:
I did try to do this but it would difficult as there would be a lot of characters that don't fit well into expect string. This is covered implicitly by checking for the presence or absence of the ceased date.

Reply:
yes, I had a look and you'd have to include html in the text which is not ideal. So leave it as it is.